File: Lab2Task1.py
def caesar_decrypt(cipher_text, shift):
    decrypted_text = ''
    for char in cipher_text:
        if char.isalpha():
            shifted_char = chr((ord(char) - shift - 65) % 26 + 65) if char.isupper() else chr((ord(char) - shift - 97) % 26 + 97)
            decrypted_text += shifted_char
        else:
            decrypted_text += char
    return decrypted_text

def brute_force_caesar(cipher_text):
    for shift in range(26):
        decrypted_text = caesar_decrypt(cipher_text, shift)
        print(f"Shift {shift}: {decrypted_text}")

cipher_text = "odroboewscdrolocdcwkbdmyxdbkmdzvkdpybwyeddrobo"
brute_force_caesar(cipher_text)

File: Lab2Task1.cpp
#include <iostream>
#include <string>

using namespace std;

string ceasar_cipher(string cipher_text, int shift) {
    string decrypted_text = "";
    for (char& c : cipher_text) {
        if (isalpha(c)) {
            char base = isupper(c) ? 'A' : 'a';
            decrypted_text += ((c - base - shift + 26) % 26) + base;
        } else {
            decrypted_text += c;
        }
    }
    return decrypted_text;
}


int main() {
    string cipher_text = "odroboewscdrolocdcwkbdmyxdbkmdzvkdpybwyeddrobo";
    int shift = 3;
    string decrypted_text = ceasar_cipher(cipher_text, shift);
    cout << "Decrypted Text = " << decrypted_text << endl;
    return 0;
}
